### PR TITLE
dpdk: update to version 23.11

### PR DESCRIPTION
**Pull request by rakshaa2000**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?

###### Change Log  

- Bump dpdk to 23.07

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Pipeline build id: [Validating](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=477629&view=results)

### CODE REVIEW

**Review by reubeno — COMMENTED** [review]
There looks to be a [23.11 release of dpdk](https://doc.dpdk.org/guides/rel_notes/release_23_11.html). Is there a reason we're not directly upgrading to that?

**Comment by rakshaa2000** [comment]
> There looks to be a [23.11 release of dpdk](https://doc.dpdk.org/guides/rel_notes/release_23_11.html). Is there a reason we're not directly upgrading to that?

I think I missed that. Will update to 23.11 instead

File: SPECS/dpdk/dpdk.signatures.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
  "Signatures": {
-  "dpdk-21.11.2.tar.xz": "10a250531090040bcc3853c2f7185ca454d536ad8d5d4b643aa8a4ae243b7117"
+  "dpdk-23.11.tar.xz": "64fa58fdfc9e9510e8e414e3bedd165bd3d4ca465a231b280323f83cd53fd865"
  }
-}
+}
```
Comment (dmcilvaney) [inline SPECS/dpdk/dpdk.signatures.json:5]:
```suggestion
}

```

**Review by dmcilvaney — APPROVED** [review]
Please fix trailing /n, but looks good otherwise.

**Pull request merged**